src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountingExtractCreateCollectorFileServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package edu.cornell.kfs.concur.batch.service.impl;
+
+import org.kuali.kfs.coreservice.framework.parameter.ParameterService;
+import org.kuali.kfs.gl.batch.CollectorBatch;
+import org.kuali.kfs.sys.service.UniversityDateService;
+import org.kuali.rice.core.api.datetime.DateTimeService;
+
+import edu.cornell.kfs.concur.batch.businessobject.ConcurStandardAccountingExtractFile;
+import edu.cornell.kfs.concur.batch.service.ConcurStandardAccountingExtractCreateCollectorFileService;
+import edu.cornell.kfs.concur.batch.service.ConcurStandardAccountingExtractValidationService;
+import edu.cornell.kfs.sys.CUKFSConstants;
+import edu.cornell.kfs.sys.CUKFSParameterKeyConstants;
+
+public class ConcurStandardAccountingExtractCreateCollectorFileServiceImpl
+        implements ConcurStandardAccountingExtractCreateCollectorFileService {
+
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(
+            ConcurStandardAccountingExtractCreateCollectorFileServiceImpl.class);
+
+    protected ConcurStandardAccountingExtractValidationService concurSAEValidationService;
+    protected UniversityDateService universityDateService;
+    protected DateTimeService dateTimeService;
+    protected ParameterService parameterService;
+
+    @Override
+    public boolean buildCollectorFile(ConcurStandardAccountingExtractFile saeFileContents) {
+        ConcurStandardAccountingExtractCollectorBatchBuilder builder = new ConcurStandardAccountingExtractCollectorBatchBuilder(
+                universityDateService::getFiscalYear, dateTimeService::toString,
+                concurSAEValidationService::validateConcurStandardAccountingExtractDetailLine, this::getConcurParameterValueAsString);
+        
+        CollectorBatch collectorBatch = builder.buildCollectorBatchFromStandardAccountingExtract(0, saeFileContents);
+        if (collectorBatch == null) {
+            LOG.error("There was a problem preparing the data for the Collector file; will not create a file.");
+            return false;
+        }
+        
+        return true;
+    }
+
+    protected String getConcurParameterValueAsString(String parameterName) {

```

Comment:
Thanks for checking, Nancy. I'll add comments accordingly so that the future consolidation work will clean this up.